packages/components/toggle-button/src/toggle-button.ts

```diff
@@ -0,0 +1,128 @@
+import { type Icon } from '@sl-design-system/icon';
+import { type EventEmitter, EventsController, event } from '@sl-design-system/shared';
+import { type SlToggleEvent } from '@sl-design-system/shared/events.js';
+import { type CSSResultGroup, LitElement, PropertyValues, type TemplateResult, html } from 'lit';
+import { property, queryAssignedElements } from 'lit/decorators.js';
+import styles from './toggle-button.scss.js';
+
+declare global {
+  interface HTMLElementTagNameMap {
+    'sl-toggle-button': ToggleButton;
+  }
+}
+
+export type ToggleButtonSize = 'sm' | 'md' | 'lg';
+export type ToggleButtonFill = 'ghost' | 'outline';
+
+/**
+ * Let the user toggle between two states.
+ *
+ * ```html
+ *  <sl-toggle-button>
+ *    <sl-icon name="far-gear"></sl-icon>
+      <sl-icon name="fas-gear" slot="pressed"></sl-icon>
+    </sl-toggle-button>
+ * ```
+ *
+ * @slot default - The icon shown in the default state of the button
+ * @slot pressed - The icon shown in the pressed state of the button
+ */
+export class ToggleButton extends LitElement {
+  /** @internal */
+  static override styles: CSSResultGroup = styles;
+
+  // eslint-disable-next-line no-unused-private-class-members
+  #events = new EventsController(this, {
+    click: this.#onClick,
+    keydown: this.#onKeydown
+  });
+
+  /** @internal The slotted default icon. */
+  @queryAssignedElements() defaultIcon?: Icon[];
+
+  /** @internal The slotted pressed state icon. */
+  @queryAssignedElements({ slot: 'pressed' }) pressedIcon?: Icon[];
+
+  /** @internal Emits when the button item has been toggled. */
+  @event({ name: 'sl-toggle' }) toggleEvent!: EventEmitter<SlToggleEvent<boolean>>;
+
+  /** Whether the toggle-button is disabled; when set no interaction is possible. */
+  @property({ type: Boolean, reflect: true }) disabled?: boolean;
+
+  /** The variant of the toggle-button. */
+  @property({ reflect: true }) fill: ToggleButtonFill = 'ghost';
+
+  /** The state of the toggle-button. */
+  @property({ type: Boolean, reflect: true }) pressed = false;
+
+  /** The size of the toggle-button component. */
+  @property({ reflect: true }) size: ToggleButtonSize = 'md';
+
+  override connectedCallback(): void {
+    super.connectedCallback();
+
+    this.setAttribute('role', 'button');
+    this.setAttribute('aria-pressed', this.pressed.toString());
+    if (!this.hasAttribute('tabindex')) {
+      this.tabIndex = 0;
+    }
+  }
+
+  override updated(changes: PropertyValues<this>): void {
+    super.updated(changes);
+    this.removeAttribute('error');
+    if (this.pressedIcon?.length === 0) {
+      console.error('There needs to be an sl-icon in the "pressed" slot for the component to work');
+      this.setAttribute('error', 'true');
+    }
+
+    if (changes.has('size')) {
+      this.#setIconProperties();
+    }
+
+    if (this.pressedIcon?.[0]?.name === this.defaultIcon?.[0]?.name) {
+      console.error(
```

Comment:
I'm not sure we do this anywhere yet, but perhaps only do this when NOT in production? I'm not sure we should do this in *this* PR though. And it requires build changes as well. From Lit: 
![CleanShot 2024-07-10 at 14 47 21@2x](https://github.com/sl-design-system/components/assets/3968/818812ae-5c1c-40b9-8e33-e49760569a26)

Reply:
In Icon we have something similar: 
<img width="1035" alt="image" src="https://github.com/sl-design-system/components/assets/1351956/669bc58c-fe79-47ce-84d7-762b241167d9">